packages/smooth_app/lib/pages/scan/smooth_barcode_scanner_mlkit.dart

```diff
@@ -0,0 +1,239 @@
+    torchEnabled: false,
+    formats: _barcodeFormats,
+    facing: CameraFacing.back,
+    detectionSpeed: DetectionSpeed.normal,
+    detectionTimeoutMs: 250, // to be raised in order to avoid crashes
+    returnImage: false,
+    autoStart: true,
+  );
+
+  Future<void> _start() async {
+    if (_isStarted) {
+      return;
+    }
+    try {
+      await _controller.start();
+      _isStarted = true;
+    } on Exception catch (e) {
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(
+          content: Text('Something went wrong (start)! $e'),
+          backgroundColor: Colors.red,
+        ),
+      );
+    }
+  }
+
+  Future<void> _stop() async {
+    if (!_isStarted) {
+      return;
+    }
+    try {
+      await _controller.stop();
+      _isStarted = false;
+    } on Exception catch (e) {
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(
+          content: Text('Something went wrong (stop)! $e'),
+          backgroundColor: Colors.red,
+        ),
+      );
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) => VisibilityDetector(
+        key: const ValueKey<String>('VisibilityDetector'),
+        onVisibilityChanged: (final VisibilityInfo info) async {
+          if (info.visible) {
+            if (_visible) {
+              return;
+            }
+            _visible = true;
+            await _start();
+            return;
+          }
+          if (!_visible) {
+            return;
+          }
+          _visible = false;
+          await _stop();
+        },
+        child: LayoutBuilder(
+          builder: (BuildContext context, BoxConstraints constraints) {
+            final Rect scanWindow = Rect.fromCenter(
+              center: Offset(
+                constraints.maxWidth / 2,
+                constraints.maxHeight / 2,
+              ),
+              width: constraints.maxWidth - 2 * MINIMUM_TOUCH_SIZE,
+              height: constraints.maxHeight,
+            );
+            return Stack(
+              children: <Widget>[
+                MobileScanner(
+                  controller: _controller,
+                  fit: BoxFit.cover,
+                  scanWindow: scanWindow,
+                  errorBuilder: (
+                    BuildContext context,
+                    MobileScannerException error,
+                    Widget? child,
+                  ) =>
+                      EMPTY_WIDGET,
+                  onDetect: (final BarcodeCapture capture) async {
+                    for (final Barcode barcode in capture.barcodes) {
+                      final String? string = barcode.displayValue;
+                      if (string != null) {
+                        await widget.onScan(string);
+                      }
+                    }
+                  },
+                ),
+                Container(
+                  decoration: ShapeDecoration(
+                    shape: zxing.QrScannerOverlayShape(
```

Comment:
Indeed:
* zxing comes with a visor, mlkit does not, so if we use zxing's visor in mlkit we have the same UI
* of course we could create a custom visor, different from zxing's, and use it in zxing and mlkit
* to be discussed - I'm not sure it's worth to hassle to "xor" zxing and mlkit, code wise. We can just embed both, while using the same "flavors" as now and use only one barcode scanner, depending on the entry point parameter

Reply:
What about the one we had before

Reply:
> What about the one we had before

To be honest it doesn't look that different.
But if needed, please create an issue asking for the old painter from `scan_visor.dart` to be implemented, instead of the default zxing visor.